src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py

```diff
@@ -401,3 +405,213 @@ def flexible_server_version_upgrade(cmd, client, resource_group_name, server_nam
             parameters=parameters),
         cmd.cli_ctx, 'Updating server {} to major version {}'.format(server_name, version)
     )
+
+
+# Custom functions for identity
+def flexible_server_identity_assign(cmd, client, resource_group_name, server_name, identities):
+    instance = client.get(resource_group_name, server_name)
+    if instance.replication_role == 'Replica':
+        raise CLIError("Cannot assign identities to a server with replication role. Use the primary server instead.")
+
+    identities_map = {}
+    for identity in identities:
+        identities_map[identity] = {}
+
+    parameters = {
+        'identity': mysql_flexibleservers.models.Identity(
+            user_assigned_identities=identities_map,
+            type="UserAssigned")}
+
+    if isinstance(client, MySqlServersOperations):
+        replica_operations_client = cf_mysql_flexible_replica(cmd.cli_ctx, '_')
+    else:
+        # pending postgres
```

Comment:
I was just curious if you know the identity assign logic for PG will use the same flow as mysql (== not need to separate these functions into each mysql and pg files?). 

I'm seeing these differences haven't thought  too much about it thou. lmk what you think 
` parameters = {
        'identity': mysql_flexibleservers.models.Identity(`
Or replicas (I assume PG replica is not ready) 

Properties in identities might also be different.

Reply:
Logic flow is the same but maybe the structure of the objects will be different, for both identity and ad-admin. For now, I think I'll leave these functions here in flexible_server_custom_common.py and when I need to implement them for PG, I'll check how similar the objets are: if they differ too much, I'll separate the functions in their respective files.